### PR TITLE
Add a tool with Sonar C++ analysis.

### PR DESCRIPTION
**Pull request by frederic-tingaud-sonarsource**
This is a preview of the static analysis capabilities of the Sonar C++ analyzer. See https://www.sonarsource.com/ for more information.

### CODE REVIEW

File: lib/tooling/sonar-tool.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2022, Compiler Explorer Authors
```
Comment (mattgodbolt) [inline lib/tooling/sonar-tool.ts:1]: It's 2023 now, but I don't think that's a blocker :)

File: lib/tooling/sonar-tool.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+    constructor(toolInfo: ToolInfo, env: ToolEnv) {
+        super(toolInfo, env);
+
+        this.addOptionsToToolArgs = false;
+        this.parseOutput = this.parseOutputAndFixes;
+    }
+
+    makeURL(rule: string): string {
+        return `https://rules.sonarsource.com/cpp/RSPEC-${rule.substring(1)}`;
```
Comment (mattgodbolt) [inline lib/tooling/sonar-tool.ts:56]: so cool!

**Review by mattgodbolt — APPROVED** [review]
Looks good: I think the only way to test it is on the site, so I will trust your local testing and merge!

As discussed in https://github.com/compiler-explorer/infra/pull/958 there are some minor tweaks to make to the tarball in future but this is a great start :) Thank you!

**Comment by apmorton** [comment]
It seems like the icon bit of this isn't working right - getting sentry errors:

```
Error: Cannot find module './Sonar.svg'
```

@frederic-tingaud-sonarsource  can you take a look?

**Comment by partouf** [comment]
I did see the icon, could it be a browser specific issue?

**Comment by apmorton** [comment]
Are you on dark theme maybe?

**Comment by partouf** [comment]
yes, but light works too

**Comment by jeremy-rifkin** [comment]
This is live

**Comment by frederic-tingaud-sonarsource** [comment]
@apmorton do you still have the icon error? I can't see anything wrong on my side.

**Comment by mattgodbolt** [comment]
I've seen some reports in our error handler about this:

```
Error: Cannot find module './Sonar.svg'
```

is what gets logged. Unfortunately no stack trace :(

**Pull request merged**